lib/impl/AdobeStateStore.js

```diff
@@ -0,0 +1,377 @@
+   * @returns {string} the constructed request url
+   */
+  createRequestUrl (key, queryObject = {}) {
+    let requestUrl
+
+    if (key) {
+      requestUrl = new url.URL(`${this.endpoint}/v1/containers/${this.namespace}/${key}`)
+    } else {
+      requestUrl = new url.URL(`${this.endpoint}/v1/containers/${this.namespace}`)
+    }
+
+    requestUrl.search = (new url.URLSearchParams(queryObject)).toString()
+    return requestUrl.toString()
+  }
+
+  /**
+   * Get Authorization headers.
+   *
+   * @private
+   * @returns {string} the authorization headers
+   */
+  getAuthorizationHeaders () {
+    return {
+      Authorization: `Basic ${this.apikey}`
+    }
+  }
+
+  /**
+   * Instantiates and returns a new AdobeStateStore object
+   *
+   * @static
+   * @param {AdobeStateStoreCredentials} credentials the credential object
+   * @returns {Promise<AdobeStateStore>} a new AdobeStateStore instance
+   * @memberof AdobeStateStore
+   * @override
+   * @private
+   */
+  static async init (credentials = {}) {
+    // include ow environment vars to credentials
+    if (!credentials.namespace && !credentials.apikey) {
+      credentials.namespace = process.env.__OW_NAMESPACE
+      credentials.apikey = process.env.__OW_API_KEY
+    }
+
+    const cloned = utils.withHiddenFields(credentials, ['apikey'])
+    logger.debug(`init AdobeStateStore with ${JSON.stringify(cloned, null, 2)}`)
+
+    const validation = joi.object().label('adobe').keys({
+      apikey: joi.string(),
+      namespace: joi.string()
+    }).required()
+      .validate(credentials)
+    if (validation.error) {
+      logAndThrow(new codes.ERROR_BAD_ARGUMENT({
+        messageValues: [validation.error.message],
+        sdkDetails: cloned
+      }))
+    }
+
+    return new AdobeStateStore(credentials.apikey)
+  }
+
+  /* **************************** ADOBE STATE STORE OPERATORS ***************************** */
+
+  /**
+   * Retrieves the state value for given key.
+   * If the key doesn't exist returns undefined.
+   *
+   * @param {string} key state key identifier
+   * @returns {Promise<AdobeStateStoreGetReturnValue>} get response holding value and additional info
+   * @memberof AdobeStateStore
+   */
+  async get (key) {
+    validateKey(key, { key })
+    logger.debug(`get '${key}'`)
+    return this._get(key)
+  }
+
+  /**
+   * Creates or updates a state key-value pair
+   *
+   * @param {string} key state key identifier
+   * @param {any} value state value
+   * @param {AdobeStateStorePutOptions} [options={}] put options
+   * @returns {Promise<string>} key
+   * @memberof AdobeStateStore
+   */
+  async put (key, value, options = {}) {
+    const details = { key, value, options }
+    validateKey(key, details)
+    validateValue(value, details)
+    validateInput(options, joi.object().label('options').keys({ ttl: joi.number() }).options({ convert: false }), details)
+
+    const ttl = options.ttl || AdobeStateStore.DefaultTTL // => undefined, null, 0 sets to default
```

Comment:
Should we omit the DefaultTTL here as it is already defined in the server?

Reply:
good idea, will change

Reply:
done